polyply/src/gen_seq.py

```diff
@@ -56,24 +56,14 @@ def _random_replace_nodes_attribute(graph, residues, weights, attribute, seeds=N
     weights: list[float]
     attribute: str
         the attribute to be replaced
-    seed: list[float]
-        a list of seeds to use in random selection
-        there needs to be one seed per node
+    seed: float
 
     Returns:
     --------
     `:class:networkx.Graph`
     """
-    if not seeds:
-       seeds = [None for _ in graph.nodes]
-
-    if len(seeds) != len(graph.nodes):
-       msg=("Too few seeds. The number of seeds must be the same as "
-            "the number of nodes in the graph.")
-       raise IOError(msg)
-
-    for node, seed in zip(graph.nodes, seeds):
-        random.seed(seed)
+    random.seed(seed)
```

Comment:
I would still move the seeding to the test, but meh.
And what if seed is None? Does that still affect the RNG?

Reply:
I think I'd prefer to have a seed option there so that you can use that in a CLI without having to set RNG somewhere far from the function. 

If seed is None, it will simply initialize the RNG using the current time or whatever else variables go in the RNG.